jmclient/jmclient/payjoin.py

```diff
@@ -903,11 +904,21 @@ def render_POST(self, request):
                               self.manager.change_out.scriptPubKey))}
 
         # we now know there were one/two outputs and know which is payment.
-        # bump payment output with our input:
+        # set the ordering of the outputs correctly.
         if change_out:
-            outs = [pay_out, change_out]
+            # indices of original payment were set in JMPayjoinManager
+            # sanity check:
+            if self.manager.change_out_index == 0 and \
+               self.manager.pay_out_index == 1:
+                outs = [change_out, pay_out]
+            elif self.manager.change_out_index == 1 and \
+                 self.manager.pay_out_index == 0:
+                outs = [pay_out, change_out]
+            else:
+                assert False, "More than 2 outputs is not supported."
```

Comment:
I think TODO comment should be added and error changed to more user friendly message (log instead of assert?), as this is incompatibility with BIP78.

Reply:
I remember thinking about this: in fact, a more-than-two output transaction will trigger an error earlier, when the initial payment psbt is received, in the sanity check, on this line:

https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/e6bc0c1a89e3e5f01914bcefa9e020218d397439/jmclient/jmclient/payjoin.py#L230

... and this means that it makes sense to use an `assert` on the line you commented here, the associated message is just a sort of offhand message, it's only a sanity check and logically shouldn't get hit.

And the assert above, in the line I just referenced, will get caught as one of several types of Exceptions that can get thrown, here:

https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/e6bc0c1a89e3e5f01914bcefa9e020218d397439/jmclient/jmclient/payjoin.py#L827-L834

Reply:
Yes, you are right. I even remember commenting out that try / except to debug some issue with receiving payjoin from Wasabi.